code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -94,25 +109,45 @@ public Account createAccount(String name, String kind, String externalId, String
         if(StringUtils.isNotEmpty(externalType)) {
             properties.put(ACCOUNT.EXTERNAL_ID_TYPE, externalType);
         }
-        Account account = resourceDao.createAndSchedule(Account.class, objectManager.convertToPropertiesFor(Account.class, properties));
-        return objectManager.reload(account);
+        return resourceDao.createAndSchedule(Account.class, objectManager.convertToPropertiesFor(Account.class, properties));
+    }
+
+    @Override
+    public Account createProject(String name, String description) {
+        Map<Object, Object> properties = new HashMap<>();
+        if (name != null) {
+            properties.put(ACCOUNT.NAME, name);
+        }
+        if (description != null) {
+            properties.put(ACCOUNT.DESCRIPTION, description);
+        }
+        properties.put(ACCOUNT.KIND, ProjectConstants.TYPE);
+        return resourceDao.createAndSchedule(Account.class, objectManager.convertToPropertiesFor(Account.class, properties));
+    }
+
+    public Account createDefaultProject(Account account) {
+        Map<Object, Object> properties = new HashMap<>();
```

Comment:
What do you mean by this exactly your wording is a little confusing. and what do you mean "merge blocker"